### ISSUE TITLE
"Unable to decrypt: The sender's device has not sent us the keys for this message." (The UISI bug)

### ISSUE THREAD

**Issue by richvdh** [vector-im/element-web#2996]
This message (or, less often, the closely related "OLM.UNKNOWN_MESSAGE_INDEX") can be caused by a number of things. This bug serves as a reference to the reasons we know about.

* You logged into your device, or joined the room, after the message was sent.  This is sort-of by design, though vector-im/element-web#2258, vector-im/element-meta#647 and vector-im/element-web#2713 are relevant.
* The sender has blacklisted your device. This is *definitely* by design, though see vector-im/element-web#3831 for improving visibility on this.
* The keys haven't arrived yet. Patience you must have.

Client-specific bugs:

 - [x] ~~#2325~~ [olm is racy when riot-web is used in multiple tabs]
 - [ ] vector-im/element-web#3309 (also on android/ios) [We can still throw away one-time keys which have messages in-flight]
 - [ ] vector-im/element-web#2783 (also on android/ios) [prioritise established olm sessions over half-open ones]
 - [ ] vector-im/element-web#3231 (also on android/ios) [different devices with the same curve25519 key confuse the olm layer]
 - [ ] https://github.com/vector-im/riot-web/issues/4216 / https://github.com/vector-im/riot-android/issues/1289 / https://github.com/vector-im/riot-ios/issues/1256 [We throw away all our one-time-keys and replace them with new ones]
 - [x] ~~#4983~~ / matrix-org/matrix-ios-sdk/issues/340 / vector-im/riot-android#1603 [yet another device list sync bug]
 - [x] ~~#5001~~ (fixed riot-web 0.12.5) [race in decryption]
 - [x] ~~#2672~~ / ~~#2305~~ / ~~#3796~~ / ~~vector-im/riot-ios#955~~ (fixed riot-ios 0.5.0) / ~~vector-im/riot-android#863~~ (fixed riot-android 0.6.10) [racy device list sync]
 - [x] ~~#2782~~ [process incoming messages before uploading new keys]
 - [x] ~~#2273~~
 - [x] ~~#2562~~
 - [x] ~vector-im/riot-android#799~ (fixed, but you may still have broken Olm sessions because of it)
 - [x] ~vector-im/riot-android#1209~ (fixed in riot-android 0.6.10, but you may still have broken Olm sessions because of it)
 - [ ] https://github.com/matrix-org/matrix-js-sdk/issues/1866
 - [x] Might not be using available session backups for some reason, and we don't re-check them when attempting to fix UISIs vector-im/element-web#20461 vector-im/element-web#21026

Protocol/server things:
 - [x] vector-im/element-web#3187 
 - [ ] matrix-org/synapse#2165
 - [x] vector-im/element-web#3754
 - [x] vector-im/element-web#3822
 - [x] vector-im/element-web#3868 
 - [ ] vector-im/element-web#3846 [sometimes the sending device doesn't seem to know about some users in the room] (hopefully will be fixed by state resets)?
 - [x] ~vector-im/riot-android#1208~ (fixed in synapse v0.21.0-rc3)
 - [x] ~matrix-org/synapse#2418 (joins can seed from servers with stale state)~
 - [ ] vector-im/element-web#6989 Device lists can get out of sync

Unexplained: 
 - [x] vector-im/element-web#2711

**Comment by ara4n** [comment]
For anyone reading this looking for a workaround, the best advice currently is that if you suddenly find yourself unable to decrypt messages from someone, ask them to open up your contact details on their client.  This should force their Riot to resync its copy of your device list, increasing the chance that the next message sent will actually be encrypted for your current device.

If this doesn't work, you have no choice but create a new room (or worst case, export your session keys, logout, login and import your session keys), until the workaround in #3553 is implemented (or this meta-bug is finally closed up)

**Comment by lampholder** [comment]
Making forward progress on this requires either @richvdh to hop back into it, or for him to hand over to somebody else.

**Comment by ara4n** [comment]
I just spent a while reviewing all of the known remaining UISI causes with @richvdh.  UISI bugs fall into two broad categories:

Wedged olm sessions:
  * #3309: Olm sessions can get wedged if we throw away OTKs for messages in-flight
  * #2783: We can reduce the risk of #3309 by prioritising established olm sessions over half-open ones
  * #2325: Olm sessions corrupt & wedge if you open the same Riot in multiple tabs
    * Fixable without much crypto-fu by swapping the storage layer for indexeddb.  Would also help mitigate #3660 (running out of local storage)
  * #3231: If you reuse node-localstorage and share curve25519 keys between different device IDs, Olm wedges <-- NOT A BUG FOR RIOT.

Missing megolm keys:
  * #2165: Caused by your HS federation being broken when you login.
    * There is no solution to this; if alice doesn't know bob's new device exists when encrypting messages for him, he'll never be able to decrypt them.
  * #3754: Caused by your HS federation being broken when someone starts a new megolm session to you, but you then receive messages from other HSes before you receive your megolm keys
    * We could query the origin HS (if it's available again) for the keys, to speed up the recovery process.
  * #3187: You ran out of Olm OTKs, so can't start a new Olm session, and thus share Megolm keys (or do anything else)
  * #3796: If Alice adds a new device whilst we're downloading her old device list, we may not spot the new device.
  * ...possibly other bugs where the act of loading the target's MemberInfo kicks the sender into refreshing their device list?

Rich estimates UISIs to roughly be a 50/50 split between the two.

However, all of the 'missing megolm key' class of bugs can be worked around by giving users a way of recovering missing megolm keys.  In turn, if we had a way of recovering missing megolm keys, we'd also have a way to share history to new devices - the infamous 'share history' bug (#2286).

Therefore the suggestion is to focus entirely[1] on solving the problem of sharing megolm keys, given the value of solving both the 'missing megolm key' bugs as well as the 'sharing history' feature is greater than the value of solving the individual bitty 'wedged olm session' bugs (which all have different solutions).  This means setting aside UISI hell and forging ahead and solving history sharing (#2286) or at least a subset of it.  This could well include improving the UX for sharing history (#2286) by supporting cross-signed keys (#2714).

[1] We can probably progress the 'multiple tabs' problem (#2325) in parallel.  And the plan is to finish the devicelist race #3797 first.

**Comment by richvdh** [comment]
> #3231: If you reuse node-localstorage and share curve25519 keys between different device IDs, Olm wedges <-- NOT A BUG FOR RIOT.

Well, it is a bug for riot, in that if anyone uses the js-sdk in the obvious manner, riot fails to talk e2e with the resultant client. It's arguable whose fault that is - ideally both ends would be fixed. But it's not a bug for riot inasmuchas it doesn't affect riot<->riot comms.

> ...possibly other bugs where the act of loading the target's MemberInfo kicks the sender into refreshing their device list?

AFAIK the only thing that loading the MemberInfo would solve these days is #3796.

**Comment by ara4n** [comment]
writing https://github.com/vector-im/riot-web/issues/2286#issuecomment-299605919 made me realise that perhaps we can also improve the experience here in general with better error messages.  For instance, do we have any way of detecting when an Olm session has got wedged, such that we can complain about that (and perhaps reset it?) rather than just whine about missing megolm keys?

**Comment by richvdh** [comment]
Yes, we can certainly improve this. We can give the user feedback about failing to decrypt to_device messages (though they tend to get replayed at initial sync, so we'd have to think how to avoid false positives). https://github.com/vector-im/riot-android/issues/800 randomly, covers that. If we can get it *reliable*, we can start a new Olm session to try and unwedge things. We can also consider giving better feedback from the sender's end (https://github.com/vector-im/riot-web/issues/2494). 

In general it's hard to tell the cause of any particular UISI, because you can't correlate it to a to_device you couldn't decrypt.

**Comment by pik** [comment]
> If users restore a device backup or tab whilst the existing device is still around and active, Olm will wedge.
> If you reuse node-localstorage and share curve25519 keys between different device IDs, Olm wedges <-- NOT A BUG FOR RIOT<->RIOT.

While it is possible at the moment I don't really like the idea of the same keys on multiple devices (this is surely a security risk). Notably also device_keys changing while device_id remains the same is another way to wedge olm sessions (the client sharing the megolm session key thinks they sent keys to the associated device, but that device has new keys and cannot decrypt the olm session).

**Comment by richvdh** [comment]
@pik: please take your questions about #3231 and #3822 to the relevant bugs.

Another thing we should consider on this bug is a way for the recipient to distinguish "the sender failed to send to you" vs "the sender chose not to send to you" (either they blocked you or your device explicitly, or because they had the 'Never send encrypted messages to unverified devices from this device' setting (https://github.com/matrix-org/matrix-js-sdk/pull/336) checked). 

- basically we should be able to distinguish between "it went a bit wrong" and "expected behaviour".

Of course that would probably mean the sender sending a "you're blocked' notification to the recipient, but that wouldn't be hard.

**Comment by richvdh** [comment]
Added #3845 for the "blocked vs failure" sidebar

**Comment by ubmarco** [comment]
The workaround 'opening the contact details' did not work for us. I can read all messages from the mobile of my colleague but not from Linux Riot. We found a workaround: my colleague closed Riot, deleted the config folder ~/.config/Riot and started the app again; now his device showed up as unverified but I could read the messages again; after verification, we get the green lock and everything works as expected.

**Comment by gerroon** [comment]
> For anyone reading this looking for a workaround, the best advice currently is that if you suddenly find yourself unable to decrypt messages from someone, ask them to open up your contact details on their client. This should force their Riot to resync its copy of your device list, increasing the chance that the next message sent will actually be encrypted for your current device.
> 
> If this doesn't work, you have no choice but create a new room (or worst case, export your session keys, logout, login and import your session keys), until the workaround in #3553 is implemented (or this meta-bug is finally closed up)

None of this works forme me:(  Creating new room, deleting the user from device and resigning in, trusting all the users oif the room from scratch etc does not work. And I have this issue with only one user that I interact with. The other user is using Android (Fdroid) and Linux versions.


I am wondering if anyone else was completely got rid of this issue without deleting the user from Synapse.

**Comment by dbkr** [comment]
Opened https://github.com/vector-im/riot-web/issues/13473 as a specific type of UISI that seems to be cropping up

**Comment by jlusPrivat** [comment]
I had long troubles finding out, that the option "never send encrypted messages from this session to unverified sessions" was causing troubles... In hindsight it was pretty obvious, that it didn't work, nevertheless it cost me many hours thinking it was my HS config.

**Comment by aaronraimist** [comment]
@jlusPrivat What version of Element are you using? The current version should show a different message when that option is enabled.

Instead of saying "Unable to decrypt: The sender's device has not sent us the keys for this message." it should say "Unable to decrypt: The sender has disabled encrypting to unverified devices."

**Comment by jlusPrivat** [comment]
I use 1.7.16 (latest) for the windows as well as the browser variant (from element.io). The android version (1.0.13) showed "Waiting for this message. This could take a while" (spontaniously translated from german version). That would've been neat, if the message showed, that the sender disabled encryption to unverified devices.

**Comment by aaronraimist** [comment]
I don't know anything about Android but the web version definetly should show different messages. I thought it might be a translation issue where someone just used the same translation but those messages don't appear to be translated at all.
<img width="677" alt="screenshot" src="https://user-images.githubusercontent.com/5855073/103691669-28801400-4f5c-11eb-9c58-b1f2ce36ea33.png">

Are you sure you didn't see these messages? Can you test again? They look very similar so you may have just stopped reading at "Unable to decrypt". If it doesn't say "Unable to decrypt: The sender has disabled encrypting to unverified devices" when that option is enabled then you should file a separate bug.

**Comment by Mjk29** [comment]
Started getting this message a couple hours ago using the Android client, as well as the windows client. 
`** Unable to decrypt: The sender's device has not sent us the keys for this message. **`

Saw the same error on 1/3/2021 after an update, resolved by verifying the other party.

**Comment by jungletek** [comment]
> 
> 
> I had long troubles finding out, that the option "never send encrypted messages from this session to unverified sessions" was causing troubles... In hindsight it was pretty obvious, that it didn't work, nevertheless it cost me many hours thinking it was my HS config.

That option is terribly unintuitively named / described. Would love to see that changed, clarified. The way it is now, I and other well-meaning security-conscious new users read it as doing the opposite, FWIW.

**Comment by stephen304** [comment]
In addition to having a better error message, I think it would be a good solution to warn the message sender the first time they send a message in a room if their setting will prevent their message from being read by people in the room. I'm in a large room and it's just pages and pages of UTD as new users come in and we have to figure out whether it's just lagging or whether that person didn't change the setting. The room admins set the room topic telling everyone to change the setting, which is a really annoying thing to have to do, especially since a good portion of people don't seem to notice it.

**Comment by aaronraimist** [comment]
@stephen304 what setting are you talking about? "Never send encrypted messages to unverified sessions from this session"? Why do they have to tell everyone to change it? It is not on by default.

**Comment by stephen304** [comment]
> @stephen304 what setting are you talking about? "Never send encrypted messages to unverified sessions from this session"? Why do they have to tell everyone to change it? It is not on by default.

Beats me. Last time I created an account was years ago, all I know is tons of people are joining because of recent events and it's a mess. If it's not on by default then maybe adding a feature that reminds people that half of the room can't read their messages would be a good idea. Because it seems like a ton of people are unaware.

**Comment by weeman1337** [comment]
I am struggling with that, too at the moment.

I have a desktop Element client showing the message. Clicking re-request ends up in an incoming key request on my mobile client. But the request then is stuck in "pending" state. I don't know whether that is already covered by an issue listed in the description.

**Comment by mattgphoto** [comment]
Running into this on a new account, as well from Desktop Client, which is the only client I'm logged into and verified on.

**Comment by masscream** [comment]
I also experience this bug on my two synapse instances. Today I wanted to send the debug logs but seems that I can't - I'm getting error: Failed to send logs: HTTP 500. Any hints or progress on this one, please ?

**Comment by vmario89** [comment]
this bug happens for 2 years now :-( no idea to get rid

**Comment by masscream** [comment]
In the meantime, I upgraded both synapse instances to the latest version and forced to update the Element on the clients side. The new messages started coming OK, the "corrupted" ones, even sent for test purposes before the upgrade stayed hanging, probably forever. If it now stays working, I'm OK. Otherwise I would probably turn off the encryption and its enforcing preference from the new rooms...

**Comment by dkasak** [comment]
To be clear, this is an umbrella issue with many possible different causes, not a narrow problem that's not being worked on. Most of the time when you're unable to decrypt a message due to a bug, you'll get such an error message displayed.

Therefore, your UISI might not have the same cause as the next one. Many causes of UISIs have already been solved and solving the remaining ones is currently a topic in active focus.

**Comment by lachesis** [comment]
My network is moving back from Element to Signal as a result of this bug. In particular, my messages sent from Element Desktop often enter this state for other users running Element Android. Nobody is having any issue Desktop -> Desktop or Android -> Android, or even Android -> Desktop. Sometimes messages exit this state, but I'd say only about 1/3rd ever become readable. Most remain in this state forever.

**Comment by the-j0k3r** [comment]
@lachesis I had a feeling that was also a factor, hence I mentioned,

It also happens to me when via DM to people on different servers as I found out just yesterday,. And to regular users vai existing DM windows after logging off and back on. 

Session verification on login should be 100% one step not two two as is.

As a result I came to the conclusion that

A) Cannot force anyone to use just one platform or any particular client (shouldn't be an issue anyway) or force them to verify sessions.
B) Because of A, Encryption is a pipe dream for on matrix.org and a quick swift kick to any privacy focused communities.

I have heard on help channels, other servers dont have such issues either.

**Comment by uhoreg** [comment]
Please not that this isn't a single bug.  This issue can be caused by multiple different things.  We're investigating and have fixed some issues, but there may be other problems.

As [mentioned earlier](https://github.com/vector-im/element-web/issues/2996#issuecomment-947787079), you can help us fix issues by submitting debug logs on both the sending and receiving sides when it happens.

**Comment by the-j0k3r** [comment]
@uhoreg

Submitted logs already, thanks, and Im pretty aware there are many vectors at play, its quite obvious that different platforms are not playing well together amongst other factors.

Cant do anything else, having submitted multiple logs for different issues linking to specific bug reports. we are at the mercy of your foo.

If you need any more specific info let us know, Ill try as best possible providing such.

Good luck.

**Comment by spencePepOS** [comment]
I'm working in a build/code & development environment,  logging in from 30 or more installations of my OS per day... 

There should be a SIMPLE box to check "Yes all these logins are ME..." I'm verifying them ALL by my phone...  my samsung adroid device has no issues... but this incessant need to wait because "failed to decrypt" across my multiple installs on three different testing machines is more than an annoyance. 

disheartening to see this is an open & unresolved issue from 2017

**Comment by uhoreg** [comment]
If you are logging in, then it is expected that you will be unable to read old messages until your device has access to key backup or can receive the keys from another device.  This is not a bug; it's just the way encryption works.

> disheartening to see this is an open & unresolved issue from 2017

This is an umbrella issue.  There are many different things that cause "Unable to decrypt" messages, some of which are not bugs but are expected behaviour.  Several of the causes have been fixed over the years.  Some don't have a clear fix.

**Comment by foresto** [comment]
At least some of us following this umbrella issue appreciate that it's not easy and will take time.

Nevertheless, one form or another of this error *still* pops up somewhat often when it should not, and has been doing so for years. It creates frustrating experiences that undermine confidence in Matrix, driving new users away, and makes it very difficult to get Matrix accepted by our communities. Given the impact of the so-called "network effect", perhaps it's time to prioritize sorting out the various causes of this issue, fix the ones that are bugs, and fix the UI around the expected cases?

For my part, I have spent many hours helping less technically inclined users troubleshoot things when they go wrong, and have submitted reports and logs when possible. I have been doing my best to manage expectations among my contacts, assuring them that Matrix and its client apps are improving, and that the rough edges are being filed down, and that bugs are not being ignored. After years of frustrations, though, people are (understandably) losing patience. If these bugs aren't fixed soon, I fear people are going to give up on Matrix. Nobody wants an unreliable messaging system.

The latest This Week in Matrix had an [encouraging excerpt](https://matrix.org/blog/category/this-week-in-matrix#community-testing) about chasing down E2EE bugs in the coming week. Are the bugs behind this issue to be included in that effort?

**Comment by spencePepOS** [comment]
I just find it irritating... that the moment I finish verifying my login with my phone as a secondary device... all previous sessions, which were ALSO verified with my secondary device...

are now UNTRUSTED?

Nobody else but me is logging into this account...

All previous desktop sessions are no longer accessible... as they've been wiped away during my constant testing from within newly installed desktop instances...

Perhaps I am an unusual use case?

spence

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by uhoreg** [comment]
> The latest This Week in Matrix had an [encouraging excerpt](https://matrix.org/blog/category/this-week-in-matrix#community-testing) about chasing down E2EE bugs in the coming week. Are the bugs behind this issue to be included in that effort?

My understanding of the community testing sessions is that it has been mainly focused (so far) on testing previously-reported issues to see if they are still problems.  The bugs behind this issue tend to be difficult to reproduce, so I don't think that they will be part of the community testing sessions.

Some of the things that we have planned soon for this bug are:

- improving the display of undecryptable messages, which may include (where possible) an explanation what the user can do to try to fix it
- (optionally) automatically submitting debug logs when there is an undecryptable message.  If both the sender and receiver have the option enabled, it will automatically send both sets of logs, to help us track down the cause of the failure
- improving the logging so that we can more easily determine the cause of the failure

> For my part, I have spent many hours helping less technically inclined users troubleshoot things when they go wrong, and have submitted reports and logs when possible.

Thank you!

**Comment by uhoreg** [comment]
> I just find it irritating... that the moment I finish verifying my login with my phone as a secondary device... all previous sessions, which were ALSO verified with my secondary device... are now UNTRUSTED?

If previously-verified devices are no longer verified, that's a different bug.  Please file a new issue, and submit debugging logs from both devices that you're verifying.

**Comment by the-j0k3r** [comment]
Trying to get this to work across devices and verified sessions is like pulling teeth, Ive tried it and given up, its a lovely idea on paper but in practice its not even ready for alpha never mind beta or production.

**Comment by dkasak** [comment]
@the-j0k3r: I understand the frustration, but please let's try keeping constructive here since this is essentially a bug tracker.

It's hard to tell what's happening on your end, but your case is definitely not typical. For instance, I keep fully verified sessions with a few friends and family and relatively frequently log in from new devices, and I'm just not seeing the problems you're describing. That's why the encryption doesn't come with big, fat warnings -- because it works much more often than not.

You must be encountering some pathological case, but to identify what's wrong we must first see what you see. I can appreciate you may be beyond the point where you're interested in engaging to supply this info, but I can assure you it's not the case that the team doesn't care about this.

**Comment by spencePepOS** [comment]
The "unusual-ness" is that none of my prior sessions exist, save the one on my phone

... most exist for only a few hours, some only exist for a few minutes before I have installed a new version of my software over top the previous versions...

Therefore, there are no other sessions to request decryption nor seek keys from...

Once I verify a session with my phone, I see no reason for previous sessions to become once again unverified or untrusted..

-------- Original Message --------

**Comment by the-j0k3r** [comment]
> @the-j0k3r: I understand the frustration, but please let's try keeping constructive here since this is essentially a bug tracker.
> 
> It's hard to tell what's happening on your end, but your case is definitely not typical. For instance, I keep fully verified sessions with a few friends and family and relatively frequently log in from new devices, and I'm just not seeing the problems you're describing. That's why the encryption doesn't come with big, fat warnings -- because it works much more often than not.
> 
> You must be encountering some pathological case, but to identify what's wrong we must first see what you see. I can appreciate you may be beyond the point where you're interested in engaging to supply this info, but I can assure you it's not the case that the team doesn't care about this.

I have submitted debug logs ad nauseam linking this issue every time I encountered such issues, so your assessment is incorrect, what I will not do is keep trying to test the limits of my sanity and patience by trying the same thing over and over and expecting different results, when the results are consistently towards the negative.

Towards this end I have stated I have 100% given up on what appears to me and everyone else involved are not interested in bad UX and so all such is abandoned.

FYI Im using always Element web over Firefox with 100% verified sessions which accumulate in profile making endless large lists from same IP/device, always my Desktop, unlike others I dont use apps or android clients, which I never understood, unless different IP, but OK.

So with this in mind I can categorically and emphatically claim my experience with my own sessions over encrypted channel has failed 90% of the time.

So there two more pieces of information I'm quite happy to share, which I never said, I wouldn't, I just said that towards encryption Im done.

**Comment by uhoreg** [comment]
> FYI Im using always Element web over Firefox with 100% verified sessions which accumulate in profile making endless large lists from same IP/device,

That sounds like you're re-logging-in frequently (e.g. his can happen if you use Element in an incognito tab, or with certain privacy settings enabled, as the browser will not retain the information needed to keep you logged in).  If that's the case, then it's expected that encryption will not work for you.  This will be fixed by [dehydrated devices](https://github.com/vector-im/element-web/issues/3825), which is currently behind a labs flag.

**Comment by ShadowJonathan** [comment]
Hold on, yeah @the-j0k3r, are you using Element in an incognito tab, and constantly have to log in again and again?

**Comment by the-j0k3r** [comment]
No incognito tabs. But yes I constantly have to login again and again. I presume its due to my cookie handling of no cross site.
![Capture](https://user-images.githubusercontent.com/31389848/146213251-5fe328d7-0e6d-4f49-8b57-f1c967edc835.PNG)

I also have [x] Clear history when Firefox closes enabled.

Firefox is the latest 95, but this has happened always since... 90.x which is when I joined Matrix.

**Comment by aaronraimist** [comment]
@the-j0k3r more likely it is related to this option
<img width="694" src="https://user-images.githubusercontent.com/5855073/146214085-4a377d92-cf74-471c-aca4-424a56219b70.png">

If you don't want to disable it, you can add an exception so only Element's data will be preserved across restarts of the browser.

**Comment by the-j0k3r** [comment]
> @the-j0k3r more likely it is related to this option <img alt="" width="694" src="https://user-images.githubusercontent.com/5855073/146214085-4a377d92-cf74-471c-aca4-424a56219b70.png">
> 
> If you don't want to disable it, you can add an exception so only Element's data will be preserved across restarts of the browser.

Not at all, I dont have that enabled, as I use many sites where I am kept logged in just fine, e.g. GitHub
![Capture](https://user-images.githubusercontent.com/31389848/146215348-9d7d80e8-fcc6-4674-a872-bdd19747683f.PNG)

**Comment by aaronraimist** [comment]
I've been running with strict blocking for at least a year and have never been automatically logged out of Element. I also tried those specific custom blocking settings and was not logged out. No exceptions related to Element.

I just tested the Firefox will never remember history setting and that did cause me to be logged out. It looks like when you enable that option it basically runs all tabs in private browsing mode.
<img width="698" src="https://user-images.githubusercontent.com/5855073/146217460-ad32b279-c362-4791-8347-b258716c219e.png">
So if you have that enabled it would make sense that you get logged out. However I'm not sure why you would stay logged in to GitHub if that is the case.

An alternative to "Firefox will never remember history" is:
<img width="724" alt="Screen Shot 2021-12-15 at 09 44 37" src="https://user-images.githubusercontent.com/5855073/146217952-b2e8ba37-8f53-409c-8110-72883e9957ea.png">
which does not run all tabs in private browsing and does not cause you to be logged out.

**Comment by ShadowJonathan** [comment]
@the-j0k3r may i ask which element URL you use? `app.element.io`?

**Comment by the-j0k3r** [comment]
Yes , this https://app.element.io always.

>Also, i think that a good test would be for someone else to try to use element with exactly those blocker settings.

And browser just in case.

**Comment by richvdh** [comment]
Please could you people start a new issue? this is meant to be an umbrella bug for UISI errors, not a general discussion ground for individual problems.

**Comment by wildcard-f8** [comment]
5 years later. still the same problem

**Comment by tehoblivious** [comment]
For me this happened to me and a friend

and the wording had me go, IT MUST BE ME? 

I should restary MY CLIENT

and whatcha know 

it sent finally 

just restart client basically

**Comment by richvdh** [comment]
We're closing this in favour of https://github.com/vector-im/element-meta/issues/245, which needs a whole load of updates, but is probably still more up-to-date than this issue (and there's no point having two issues tracking the same thing)

**Comment by nituladaboneko** [comment]
wow

5y later
and this is still an unresolved issue

was excited to move our team from google hangouts to element

but alas
another poor quality open source project

**Comment by vmario89** [comment]
we are using element for 2.5 years now and for us the errors have gone completely. it works stable 99% of time. element + matrix are not a poor source project, but your opinion

**Comment by shmerl** [comment]
This is still happening (encrypted room, one user joining from iOS element client can't ready any messages with such error showing up).

**Comment by ShadowJonathan** [comment]
iOS is currently going through a rewrite, so after that (Element X) replaces the current iOS app, those issues would most likely be resolved, as it uses the more-vetted rust-based matrix SDK to do crypto.

**Comment by shmerl** [comment]
Is there some status page for Element X for iOS to track the progress?

**Comment by richvdh** [comment]
> Please keep in mind that the issues on the meta repo are not meant to be commented on

I'm not sure that's true? element-meta is used to track issues that span more than one Element client implementation. Comments on issues in it are as welcome there as in the individual repositories.

That said:
 * element-meta does have a relatively high proportion of "tracker" issues which exist only to link out to other, more granular issues. Typically discussion is more appropriate on the lower-level issues than the tracker issues.
 * comments along the lines of "why isn't this thing ready yet" aren't going to be terribly helpful in any case.

**Comment by pylapp** [comment]
Followed the following steps and it worked like a charm:
- export all my keys 
- log out
- kill the all
- log in with recovery sentence
